azkaban-common/src/main/java/azkaban/server/HttpRequestUtils.java

```diff
@@ -130,20 +132,25 @@ public static ExecutionOptions parseFlowOptions(final HttpServletRequest req)
    * </pre>
    */
   public static void filterAdminOnlyFlowParams(final UserManager userManager,
-      final ExecutionOptions options, final User user) throws ExecutorManagerException {
+      final ExecutionOptions options, final User user, final Props props) throws ExecutorManagerException {
     if (options == null || options.getFlowParameters() == null) {
       return;
     }
 
     final Map<String, String> params = options.getFlowParameters();
+    boolean hasAdminPermission = hasPermission(userManager, user, Type.ADMIN);
     // is azkaban Admin
-    if (!hasPermission(userManager, user, Type.ADMIN)) {
-      params.remove(ExecutionOptions.FLOW_PRIORITY);
+    if (!hasAdminPermission) {
       params.remove(ExecutionOptions.USE_EXECUTOR);
     } else {
-      validateIntegerParam(params, ExecutionOptions.FLOW_PRIORITY);
       validateIntegerParam(params, ExecutionOptions.USE_EXECUTOR);
     }
+    boolean disableAdminOnlyForFlowPriority = props.getBoolean(ConfigurationKeys.AZKABAN_DISABLE_ADMIN_ONLY_PERMISSION_FOR_FLOW_PRIORITY, false);
```

Comment:
Is it necessary to pass the whole props reference when only the boolean property could be sent?

Reply:
To keep it generic to enable or disable any other property in future, I am passing props.